src/wasm-v8.cc

```diff
@@ -605,6 +605,7 @@ auto ExternType::copy() const -> own<ExternType*> {
     case EXTERN_GLOBAL: return global()->copy();
     case EXTERN_TABLE: return table()->copy();
     case EXTERN_MEMORY: return memory()->copy();
+    default: assert(false);
```

Comment:
Hm, wouldn't having a branch with no return trip up other compilers?

Reply:
Funnily enough, I've added all those asserts (with the exception of the one in `src/wasm-v8.cc:2199`, which I added for completeness) to silence the no return errors in GCC:
```
error: control reaches end of non-void function [-Werror=return-type]
```
so this works fine for both: `clang` and `gcc`, at least using default `CFLAGS`.

But now that you pointed this out, I realized that this will indeed break when building with `-DNDEBUG` (i.e. when `assert()` is omitted), but this is already the case in `master`, since the same pattern is used in a few other places, e.g. in `src/wasm-bin.cc:220`:
https://github.com/WebAssembly/wasm-c-api/blob/dc8cc29340e0c374dd350e0de8ff4b22d429af1e/src/wasm-bin.cc#L220

Perhaps all those `assert(false)` should be replaced with `UNIMPLEMENTED()`? What do you think?

Reply:
Ping.

Reply:
Sounds good, though I don't think we want to abuse UNIMPLEMENTED, but define a similar UNREACHABLE. Do you mind including that with this PR?